src/notifications/notificationmanager.h

```diff
@@ -21,10 +21,46 @@
 #include <QTimer>
 #include <QSet>
 #include <QDBusContext>
+#include <QDBusConnection>
+#include <QDBusMessage>
 
 class AndroidPriorityStore;
 class CategoryDefinitionStore;
 class QSqlDatabase;
+class QDBusPendingCallWatcher;
+
+/*!
+ * \class ClientIdentifier
+ *
+ * \brief Asynchronous D-Bus client identification
+ *
+ * First makes standard GetConnectionUnixProcessID() query to D-Bus daemon.
+ * Then, if the resulting pid looks like xdg-dbus-proxy process, makes
+ * Sailfish OS specific Identify() query to the proxy in order to get
+ * details of actual client behind the proxy.
+ *
+ * Emits finished() signal when done, at which state clientPid() will return
+ * pid of the client process or -1 if client could not be identified.
+ */
+class ClientIdentifier : public QObject
+{
+    Q_OBJECT
+public:
+    ClientIdentifier(QObject *parent, const QDBusConnection &connection, const QDBusMessage &message);
+    Q_SIGNAL void finished(ClientIdentifier *what);
+    QDBusConnection &connection() { return m_connection; }
+    QDBusMessage &message() { return m_message; }
+    QString member() { return message().member(); }
+    QString clientName() { return message().service(); }
+    int clientPid() { return m_clientPid; }
+ private:
+    Q_SLOT void getPidReply(QDBusPendingCallWatcher *watcher);
+    Q_SLOT void identifyReply(QDBusPendingCallWatcher *watcher);
```

Comment:
Could declare the signals and slots the common way here?

Reply:
Oh, well... ok